src/slice_transform.rs

```diff
@@ -108,9 +108,8 @@ pub unsafe extern "C" fn in_domain_callback(
     raw_cb: *mut c_void,
     raw_key: *const c_char,
     key_len: size_t,
-) -> u8 {
+) -> c_uchar {
     let cb = &mut *(raw_cb as *mut TransformCallback);
     let key = slice::from_raw_parts(raw_key as *const u8, key_len as usize);
-    cb.in_domain_fn
-        .map_or(0xff, |in_domain| in_domain(key) as u8)
```

Comment:
Returning `0xff` here was somewhat questionable (as a `bool` is expected in C++ land), but not a problem as far as I can tell after reading the current implementation in RocksDB.